cloud/pkg/imageprepullcontroller/controller/util.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2023 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryType "k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+
+	"github.com/kubeedge/kubeedge/cloud/pkg/common/util"
+	"github.com/kubeedge/kubeedge/common/constants"
+	"github.com/kubeedge/kubeedge/pkg/apis/operations/v1alpha1"
+	crdClientset "github.com/kubeedge/kubeedge/pkg/client/clientset/versioned"
+)
+
+const ImagePrepull = "prepull"
+
+func validateNode(node *v1.Node) bool {
+	if !util.IsEdgeNode(node) {
+		klog.Warningf("Node(%s) is not edge node", node.Name)
+		return false
+	}
+
+	// if node is in NotReady state, cannot prepull images
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == v1.NodeReady && condition.Status != v1.ConditionTrue {
+			klog.Warningf("Node(%s) is in NotReady state", node.Name)
+			return false
+		}
+	}
+
+	return true
+}
+
+// buildPrepullResource build prepull resource in msg send to edge node
+func buildPrepullResource(imagePrepullName, nodeName string) string {
+	resource := fmt.Sprintf("%s%s%s%s%s%s%s", "node", constants.ResourceSep, nodeName, constants.ResourceSep, ImagePrepull, constants.ResourceSep, imagePrepullName)
+	return resource
+}
+
+func parsePrepullresource(resource string) (string, string, error) {
+	var nodeName, jobName string
```

Comment:
These variables don't seem to matter, right?

Reply:
> These variables don't seem to matter, right?

For now, this variables are used to standardize the resource of the imagePrePull message. And we will make validation after parsing it.